### PR TITLE
Automatic background color for TV logos in recording details

### PR DESCRIPTION
**Pull request by wutschel**
Implements missing function report in [forum post](https://forum.kodi.tv/showthread.php?tid=359717&pid=3027541#pid3027541).

Details:
- Choose different background colors for TV logos
- Processing only done when showing PVR recording details
- Refactoring into Utilities.m to re-use same implementation in different controllers

### CODE REVIEW

File: XBMC Remote/Utilities.m [hunk 1]
```diff
@@ -157,6 +157,30 @@ - (UIImage*)colorizeImage:(UIImage *)image withColor:(UIColor*)color{
     return img;
 }
 
++ (UIImage*)colorizeImageBackground:(UIImage*)image color:(UIColor*)newcolor {
+    CALayer *imageLayer = [CALayer layer];
+    imageLayer.frame = CGRectMake(0, 0, image.size.width, image.size.height);
+    imageLayer.contents = (id) image.CGImage;
```
Comment (kambala-decapitator) [inline XBMC Remote/Utilities.m:163]: don't add spaces to casts

File: XBMC Remote/Utilities.m [hunk 2]
```diff
@@ -157,6 +157,30 @@ - (UIImage*)colorizeImage:(UIImage *)image withColor:(UIColor*)color{
...
+    imageLayer.masksToBounds = YES;
+    imageLayer.backgroundColor = newcolor.CGColor;;
```
Comment (kambala-decapitator) [inline XBMC Remote/Utilities.m:165]: ;;

File: XBMC Remote/ShowInfoViewController.m [hunk 3]
```diff
@@ -729,6 +738,7 @@ -(void)createInfo{
     // NEED TO BE OPTIMIZED. IT WORKS BUT THERE ARE TOO MANY IFS!
     NSMutableDictionary *item = self.detailItem;
     NSString *placeHolderImage = @"coverbox_back.png";
+    isRecordingDetail = (item[@"recordingid"] != nil);
```
Comment (kambala-decapitator) [inline XBMC Remote/ShowInfoViewController.m:741]: drop brackets

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

File: XBMC Remote/Utilities.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/Utilities.m]: done

File: XBMC Remote/ShowInfoViewController.m (hunk 3 above)
Comment (wutschel) [reply XBMC Remote/ShowInfoViewController.m]: you´re having eagle´s eyes :)
I also searched for more occurances of double semicolons and fixed them with an additional commit.

File: XBMC Remote/Utilities.m (hunk 2 above)
Comment (wutschel) [reply XBMC Remote/Utilities.m]: you´re having eagle´s eyes :)
I also searched for more occurances of double semicolons and fixed them with an additional commit.

Comment (kambala-decapitator) [reply XBMC Remote/Utilities.m]: you should double-check what you commit ;)

**Pull request merged**